api/upgradeseries/upgradeseries.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2013 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package upgradeseries
+
+import (
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/api/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/core/model"
+)
+
+const upgradeSeriesFacade = "UpgradeSeries"
+
+// State provides access to the UpgradeSeries API facade.
+type State struct {
```

Comment:
Let's not call this "State". We have way too many "State" things. The general practice for newly implemented facade apis is to use "Client".